src/main/java/org/variantsync/diffdetective/tablegen/MiningResultAccumulator.java

```diff
@@ -21,26 +23,16 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.function.BiConsumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /** Accumulates multiple {@link AnalysisResult}s of several datasets. */
 public class MiningResultAccumulator {
```

Comment:
Rename to `AnalysesResultAccumulator`?

Reply:
I don't think this is a better name for the current implementation as it doesn't accumulate all results but only `ExplainedFilterSummary`, `EditClassCount` and `StatisticsAnalysis` which are created by the `DiffTreeMiner`.
We could generalize `MiningResultAccumulator` by either
1. creating a registry for all `Metadata` so it can be created in `AnalysesResultAccumulator`, or
2. adding the qualified class names to the serialization and creating the relevant classes at runtime.

Then `AnalysesResultAccumulator` would be appropriate.

Reply:
Let's discuss that in a meeting. Putting effort in here only makes sense if we actually profit from such a refactoring.